### ISSUE TITLE
`PieSectorDataItem` has incorrect type for `payload` property

### ISSUE THREAD

**Issue by BTOdell** [recharts/recharts#5261]
## Do you want to request a feature or report a bug?

Bug

## What is the current behavior?

TypeScript type is incorrect for `PieSectorDataItem.payload`.

## If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via codesandbox or similar (template: https://codesandbox.io/p/sandbox/simple-line-chart-v25dl4).

https://github.com/recharts/recharts/blob/a73a8ce24667bc982ac50e79c80781aae827688b/src/polar/Pie.tsx#L75
(`master` branch as of November 19th, 2024)

`payload` doesn't contain an array, but instead is a single payload object.

## What is the expected behavior?

I would expect the type to just be `any`. It seems that this has been fixed in the `3.x` branch:
https://github.com/recharts/recharts/blob/9557a6edf37678523881cc5f7bfdeba278f790f3/src/polar/Pie.tsx#L92

Could this change be backported to the `2.x` version? Going from `any[]` to `any` shouldn't result in broken code.

## Which versions of Recharts, and which browser / OS are affected by this issue? Did this work in previous versions of Recharts?

Recharts 2.13.3

**Comment by ckifer** [comment]
Can you paste a snippet of your usage? Where is this causing the error so I can validate a fix?

Could you reproduce in a codesandbox?
https://codesandbox.io/p/sandbox/two-simple-pie-chart-forked-f2fydz?file=%2Fsrc%2FApp.tsx%3A25%2C41

**Comment by BTOdell** [comment]
The official example code can actually be used:
https://recharts.org/en-US/examples/CustomActiveShapePieChart

But the `props` parameter needs to be typed with `PieSectorDataItem`:

```ts
import type { PieSectorDataItem } from "recharts/types/polar/Pie";

const renderActiveShape = (props: PieSectorDataItem) => {
...
```

`payload.name` on line 27 will have a compile error.

**Comment by ckifer** [comment]
Cool thanks. This is fine to be backported to master. @BTOdell do you want to make a quick PR?

**Comment by ckifer** [comment]
released in 2.14.0
